generate.py

```diff
@@ -91,39 +94,41 @@ def main(
         temperature: A value controlling the randomness of the sampling process. Higher values result in more random
             samples.
         checkpoint_path: The checkpoint path to load.
-        tokenizer_path: The tokenizer path to load.
+        config_path: The model config path to load.
+        tokenizer_path: The tokenizer vocabulary path to load.
+        tokenizer_config_path: The tokenizer config path to load.
         quantize: Whether to quantize the model and using which method:
             ``"llm.int8"``: LLM.int8() mode,
             ``"gptq.int4"``: GPTQ 4-bit mode.
     """
     if not checkpoint_path:
-        checkpoint_path = Path(f"./checkpoints/lit-llama/7B/lit-llama.pth")
+        checkpoint_path = Path(f"./checkpoints/lit-stablelm/3B/lit-stablelm.pth")
+    if not config_path:
+        config_path = Path(f"./checkpoints/lit-stablelm/3B/config.json")
     if not tokenizer_path:
-        tokenizer_path = Path("./checkpoints/lit-llama/tokenizer.model")
-    assert checkpoint_path.is_file(), checkpoint_path
-    assert tokenizer_path.is_file(), tokenizer_path
+        tokenizer_path = Path("./checkpoints/lit-stablelm/tokenizer.json")
+    if not tokenizer_config_path:
+        tokenizer_config_path = Path("./checkpoints/lit-stablelm/tokenizer_config.json")
+
+    with open(config_path) as fp:
+        config = StableLMConfig(**json.load(fp))
 
     fabric = L.Fabric(devices=1)
     dtype = torch.bfloat16 if fabric.device.type == "cuda" and torch.cuda.is_bf16_supported() else torch.float32
 
-    print("Loading model ...", file=sys.stderr)
+    print(f"Loading model: {config}", file=sys.stderr)
     t0 = time.time()
     with lazy_load(checkpoint_path) as checkpoint:
-        name = llama_model_lookup(checkpoint)
```

Comment:
Llama uses the lm head size as a proxy for the model size, but for stablelm, the same head size corresponds to different model sizes.

So instead we save the checkpoint config in `convert_hf_checkpoint.py` and use it to load reliably. Hence the added `config_path` argument in this script.